### DISCUSSION TITLE
[RFC] Convert the `LumberjackLevel` enum to a string literal union type

### DISCUSSION THREAD

**Discussion by LayZeeDK** [ngworker/lumberjack#169] [RFC]
The `LumberjackLevel` enum is essential to Lumberjack. Since it will affect many packages in the future, I think that now is the best time to propose to convert it from a TypeScript enum to a string literal union type.

## Pros
- Terser
- Decreased bundle size
- No need to import `LumberjackLevel` in most use cases
- Strong typing is maintained
- Auto-completion is maintained
- Refactoring is still supported except for a few edge cases:
   - https://github.com/microsoft/TypeScript/issues/41922
   - https://github.com/microsoft/TypeScript/issues/41923
   - https://github.com/microsoft/TypeScript/issues/41489
- Avoid a future `const enum` conversion which [can lead to issues](https://ncjamieson.com/dont-export-const-enums/)

## Cons
- Breaking change
- Less explicit
- No support for `Object.entries`, `Object.keys`, or `Object.values`.

#### Implementation
```typescript
/**
 * Level used to configure log drivers or add a log level to a log.
 */
export enum LumberjackLevel {
  Critical = 'critical',
  Debug = 'debug',
  Error = 'error',
  Info = 'info',
  Trace = 'trace',
  Verbose = 'verbose',
  Warning = 'warn',
}
```

#### Internal usage

```typescript
export const defaultProductionLevels: LumberjackConfigLevels = [
  LumberjackLevel.Critical,
  LumberjackLevel.Error,
  LumberjackLevel.Info,
  LumberjackLevel.Warning,
];
```

#### External usage

```typescript
lumberjack.log({
  level: LumberjackLevel.Info,
  message: 'Hello, World!',
  scope: 'MyComponent',
  createdAt: Date.now(),
});
```

#### Implementation
```typescript
/**
 * Level used to configure log drivers or add a log level to a log.
 */
export type LumberjackLevel = 'critical' | 'debug' | 'error' | 'info' | 'trace' | 'verbose' | 'warn';
```

#### Internal usage

```typescript
export const defaultProductionLevels: LumberjackConfigLevels = ['critical', 'error', 'info', 'warning'];
```

#### External usage

```typescript
lumberjack.log({
  level: 'info',
  message: 'Hello, World!',
  scope: 'MyComponent',
  createdAt: Date.now(),
});
```

**Comment by NachoVazquez** [comment]
I love it. It is a significant breaking change, but this is the moment for big changes.
